File: core/src/ru/myitschool/dcrawler/event/EventListener.java
package ru.myitschool.dcrawler.event;

import java.util.HashMap;

public abstract class EventListener {

    public EventListener() throws Exception {
        for (String code : eventCodes()) {
            EventController.registerEvent(code).setListener(this);
        }
    }

    protected abstract HashMap<String, Object> fire(String eventCode, HashMap<String, Object> args);

    public abstract String[] eventCodes();

    public static void initListeners() {
        try {
            new EntityEventListener();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: core/src/com/wolg_vlad/dcrawler/event/EventListener.kt
package com.wolg_vlad.dcrawler.event

import java.util.*

abstract class EventListener {
    abstract fun fire(eventCode: String?, args: HashMap<String, Any>): HashMap<String, Any>
    abstract fun eventCodes(): Array<String>

    companion object {
        fun initListeners() {
            try {
                EntityEventListener()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    init {
        for (code in eventCodes()) {
            EventController.registerEvent(code, this)
        }
    }
}